### PR TITLE
AP_RangeFinder: add LeddarTech Vu8 driver

### PR DESCRIPTION
**Pull request by rmackay9**
This adds a driver for the [LeddarTech Vu8](https://autonomoustuff.com/product/leddartech-vu8/) driver ([datasheet](https://autonomoustuff.com/wp-content/uploads/2019/10/as-leddar-vu8-20190618.pdf)).

This lidar is slightly unusual in that it provides eight distances which are taken at 2.5 degree increments in a plane giving a virtual beam width of 20degrees (other wider beam with versions are also available).  The hope is that this will work better in situations where the surface has some sharp edged bumps or depressions.  This driver will always take the shortest of the readings.

The lidar actually provides an I2C, Serial and CAN interface but I chose to implement just the serial interface.

Each sensor can be configured to use a different address using a configuration tool from the manufacturer.  Although I don't see any need for a user to change this address from the default of "1", this driver uses the RNGFNDx_ADDR parameter so users can change this to match the sensor if they've changed it for some reason.

A few things that this driver doesn't do but which we could consider for the future:

- the distances are not scaled based on their angle.  I.e. over flat ground we would expect the distances of the outer readings to be slightly longer than the central angle.  The difference is only about 1.5% though so I figured it doesn't really matter too much
- the sensor apparently provides a continuous acquisition mode but I could not figure out the format of the messages it was returning.
- the sensor provides a 2nd, 3rd, 4th, 5th and 6th detection in each direction.  We could potentially use these backup distances for something but we don't do this for other lidars that can also provide multiple readings.

This has been bench tested including cutting power to the sensor to ensure it recovers correctly.  I plan to test it on a real vehicle over the next week or so.

A connection diagram for use with a pixhawk is below.  I'll replace the pixhawk with the generic autopilot before it's put on the wiki.

leddarvu-pixhawk

### CODE REVIEW

File: libraries/AP_RangeFinder/AP_RangeFinder.cpp [hunk 1]
```diff
@@ -496,6 +497,11 @@ void RangeFinder::detect_instance(uint8_t instance, uint8_t& serial_instance)
             drivers[instance] = new AP_RangeFinder_Lanbao(state[instance], params[instance], serial_instance++);
         }
         break;
+    case Type::LeddarVu8_Serial:
+        if (AP_RangeFinder_Lanbao::detect(serial_instance)) {
```
Comment (peterbarker) [inline libraries/AP_RangeFinder/AP_RangeFinder.cpp]: This... doesn't look great :-)

```suggestion
        if (AP_RangeFinder_LeddarVu8::detect(serial_instance)) {
```

Comment (rmackay9) [reply libraries/AP_RangeFinder/AP_RangeFinder.cpp]: Thanks for that!  I guess the serial lidars all run the same detect function so it didn't actually show up in testing.  txs!

**Comment by rmackay9** [comment]
rebased on master..

**Comment by rmackay9** [comment]
I've flight tested this and it does not appear to be very reliable.  The lidar is reporting invalid distances quite often.  I suspect environmental issues with the lidar - it's very cold here or perhaps the battery voltage was dropping too low.

**Comment by rmackay9** [comment]
I've flight tested this driver now and it works.  The reliability issues are quite clearly caused by the input voltage.  If it drops below 11V the lidar is unable to provide reliable distances.

https://www.youtube.com/watch?v=EnJq63ggBao

**Comment by tridge** [comment]
LGTM

**Pull request merged**